pkg/volume/azure_file/azure_file.go

```diff
@@ -207,7 +209,12 @@ func (b *azureFileMounter) SetUpAt(dir string, fsGroup *int64) error {
 		return err
 	}
 	os.MkdirAll(dir, 0750)
-	source := fmt.Sprintf("//%s.file.core.windows.net/%s", accountName, b.shareName)
+
+	azure, err := getAzureCloud(b.plugin.host.GetCloudProvider())
```

Comment:
azure_provision.go is for dynamic provision case and that works on azure only. For user provisioned azure file share, it should work for all kinds of clusters. That's also why azure_file.go does not reference a cloud provider before.


adding feature author @rootfs, do you have any comments?